src/completions.js

```diff
@@ -9,36 +9,55 @@
 
 /* @flow */
 
-import { AbstractValue, Value } from "./values/index.js";
-import type { Effects, Realm } from "./realm.js";
+import type { BabelNodeSourceLocation } from "babel-types";
 import invariant from "./invariant.js";
+import type { Effects, Realm } from "./realm.js";
+import { AbstractValue, Value } from "./values/index.js";
 
 export class Completion {
-  constructor(value: Value, target?: ?string) {
+  constructor(value: Value, location: ?BabelNodeSourceLocation, target?: ?string) {
     this.value = value;
     this.target = target;
+    this.location = location;
   }
 
   value: Value;
   target: ?string;
+  location: ?BabelNodeSourceLocation;
 }
+
+// Normal completions are returned just like spec completions
+export class NormalCompletion extends Completion {}
+
 // Abrupt completions are thrown as exeptions, to make it a easier
 // to quickly get to the matching high level construct.
 export class AbruptCompletion extends Completion {}
-// Normal completions are returned just like spec completions
-export class NormalCompletion extends Completion {}
 
 export class ThrowCompletion extends AbruptCompletion {
-  constructor(value: Value, nativeStack?: ?string) {
-    super(value);
+  constructor(value: Value, location: ?BabelNodeSourceLocation, nativeStack?: ?string) {
```

Comment:
It sets the error location to realm.currentSourceLocation. In some cases it it will be valuable to allow callers to createErrorThrowCompletion to override that behavior. I'll add that ability when I next get to a relevant bit of code.